### ISSUE TITLE
Production-ready support of VS Code as a Che editor

### ISSUE THREAD

**Issue by l0rd** [eclipse/che#20500]
### Is your enhancement related to a problem? Please describe

Running VS Code as editor in Che workspaces

### Describe the solution you'd like

- [x] https://github.com/eclipse/che/issues/20435
- [x] Using OpenVSX instead of MS marketplace
- [x] https://github.com/eclipse/che/issues/20569 @azatsarynnyy 
- [x] https://github.com/eclipse/che/issues/20505 @benoitf 
- [x] https://github.com/eclipse/che/issues/20679 @benoitf
- [x] https://github.com/eclipse/che/issues/20508 @benoitf  
- [x] https://github.com/eclipse/che/issues/20523 @benoitf
- [x] CI to update to lastest upstream VS Code (through PR and checks) @benoitf 
- [x] https://github.com/eclipse/che/issues/20870 @benoitf
- [x] https://github.com/eclipse/che/issues/21122
- [x] VS Code init container optimization (one unique init container to run task in parallel) @benoitf 
- [x] https://github.com/eclipse/che/issues/20549 @benoitf
- [x] https://github.com/eclipse/che/issues/21443 @svor
- [x] https://github.com/eclipse/che/issues/21030 @azatsarynnyy 
- [ ] https://github.com/eclipse/che/issues/20345 @ibuziuk
- [x] https://github.com/eclipse/che/issues/21473 @azatsarynnyy 
- [x] https://github.com/eclipse/che/issues/21400 @azatsarynnyy 
- [x] https://github.com/eclipse/che/issues/21392 @azatsarynnyy 
- [ ] https://github.com/eclipse/che/issues/21431 @azatsarynnyy 
- [x] Make VS Code the default editor @ibuziuk @svor 
- [ ] Documentation updates @themr0c 
- [ ] QE: Add a happy path test using VS Code @dmytro-ndp

### Additional context

Downstream: 
- [x] [Productize che-code](https://issues.redhat.com/browse/CRW-2744)  @RomanNikitenko
- [x] [Include VSCode in OpenShift Dev Spaces (plugin-registry and samples)](https://issues.redhat.com/browse/CRW-2288) @svor  
- [ ] Multi arch support @Kasturi1820  

### Describe alternatives you've considered

Issues removed from this epic
- https://github.com/eclipse/che/issues/20507 (vscode subtasks are done)
- validate that everything works with Che single host configuration (it looks like that's working)
- https://github.com/eclipse/che/issues/20627 (that may be a nice to have but not required)
- https://github.com/eclipse/che/issues/21341 (not a requirement yet)
- Implement `ExternalURL` in VS Code (what is this for?)

**Comment by nickboldt** [comment]
Downstream: https://issues.redhat.com/browse/CRW-2288

**Comment by benoitf** [comment]
cc @sleshchenko if you can include in your next sprint
https://github.com/devfile/devworkspace-operator/issues/628
https://github.com/devfile/devworkspace-operator/issues/629

as it's blocking introduction of this editor in the plug-in registry.

**Comment by trust-freedom** [comment]
- Using OpenVSX instead of MS marketplace.
- Use the plugin registry as an alternative to OpenVSX.

So in the future, can I directly use OpenVSX marketplace to install extensions in the "Production-ready support of VS Code"? Or must che plugin mode be used?

**Comment by benoitf** [comment]
hello @trust-freedom 

There are multiple ways so maybe you can express your wishes.

Using OpenVSX directly won't work for airgap mode so we would still need the registry to embed vsix (as of today with offline image of plug-in registry). Also sometimes we may want to add some pre-defined properties

But, plug-in registry could work as a proxy to open-vsx.org (or another in-house instance) or plug-in registry could sync from a predefined/configurable list of extensions or also users could directly use a different registry (like open-vsx.org)

**Comment by azatsarynnyy** [comment]
added
- #21030

**Comment by azatsarynnyy** [comment]
added
- https://github.com/eclipse/che/issues/21122

**Comment by azatsarynnyy** [comment]
I've removed https://github.com/eclipse/che/issues/21080 from the sub-task list, as it's not relevant anymore.
